crypto/encode_decode/decoder_lib.c

```diff
@@ -755,31 +794,29 @@ static int decoder_process(const OSSL_PARAM params[], void *arg)
 
         OSSL_TRACE_BEGIN(DECODER) {
             BIO_printf(trc_out,
-                       "(ctx %p) %s [%u] Running decoder instance %p => %d\n",
+                       "(ctx %p) %s [%u] Running decoder instance %p => %d"
+                       " (recursed further: %s, constructed object: %s)\n",
                        (void *)new_data.ctx, LEVEL, (unsigned int)i,
-                       (void *)new_decoder_inst, ok);
+                       (void *)new_decoder_inst, ok,
+                       new_data.flag_next_level_called ? "yes" : "no",
+                       new_data.flag_object_constructed ? "yes" : "no");
         } OSSL_TRACE_END(DECODER);
 
-        if (ok)
+        /* Break on error */
+        if (!ok) {
+            ERR_clear_last_mark();
             break;
+        }
+        ERR_pop_to_mark();
```

Comment:
I could see one more possible improvement here - could we check if this was the last decoder invocation and there was nothing constructed yet? If so, we might want to keep the errors from the last invocation as they might be indicative.

Reply:
Have a look at the changes I just pushed

Reply:
Yeah, this looks good. So perhaps the draft status should be now lifted?